Added example directories parameter to TestCommand and Specmatic Config, aliased in StubCommand.

StarKhan6368 opened the pull request:
<!--
Thanks for your interest in the project. Bugs filed and PRs submitted are appreciated!

Please make sure that you are familiar with and follow the Code of Conduct for
this project (found in the CODE_OF_CONDUCT.md file).

Also, please make sure you're familiar with and follow the instructions in the
contributing guidelines (found in the CONTRIBUTING.md file).

Please fill out the information below to expedite the review and (hopefully)
merge of your pull request!
-->

<!-- What changes are being made? (What feature/bug is being fixed here?) -->

**What**:
-  Add `--examples` parameter to TestCommand, rename in StubCommand.
- Add `exampleDirectories` attribute to SpecmaticConfig.

<!-- Why are these changes necessary? -->

**Why**: Currently it is not possible to configure the examples directory within the test command or SpecmaticConfig.

<!-- How were these changes implemented? -->

**How**:
- Alias `examples` to `--data` argument in `StubCommand`.
- Add `--examples` in `TestCommand`.
- Refactor `loadExternalisedExamples` to use list of directories.

<!-- Have you done all of these things?  -->

**Checklist**:

<!-- add "N/A" to the end of each line that's irrelevant to your changes -->

<!-- to check an item, place an "x" in the box like so: "- [x] Documentation" -->

- [ ] Documentation added to the README.md OR link to PR on https://github.com/specmatic/specmatic-documentation N/A
- [x] Tests
- [ ] Sonar Quality Gate

<!-- feel free to add additional comments -->

StarKhan6368 committed:
Add examples arg to TestCommand, rename in Stub.

- Alias `examples` to `--data` argument.
- Add `--examples` in TestCommand.kt.
- Refactor loadExternalisedExamples to use
  list of directories.
- Add `exampleDirectories` to SpecmaticConfig.
- Add and Update tests.

StarKhan6368 committed:
Rename `exampleDirectories` to `examples`.

joelrosario commented on core/src/test/kotlin/integration_tests/LoadTestsFromExternalisedFiles.kt:
```diff
@@ -227,7 +227,7 @@ class LoadTestsFromExternalisedFiles {
         assertThat(results.successCount).isEqualTo(1)
         assertThat(results.success()).withFailMessage(results.report()).isTrue()
-        System.clearProperty(LOCAL_TESTS_DIRECTORY)
+        System.clearProperty(EXAMPLE_DIRECTORIES)
```
This should be in a finally block, both here and anywhere else this property is set. That way if the test fails, the property will get unset and will not impact any other tests.

StarKhan6368 replied:
Thanks, added logic to clear system properties in test files.

joelrosario commented on core/src/test/kotlin/io/specmatic/core/SpecmaticConfigKtTest.kt:
```diff
@@ -183,7 +188,8 @@ internal class SpecmaticConfigKtTest {
             VALIDATE_RESPONSE_VALUE to "false",
-            EXTENSIBLE_SCHEMA to "false"
+            EXTENSIBLE_SCHEMA to "false",
+            EXAMPLE_DIRECTORIES to "api/tests,api/stubs"
```
This was initially confusing as we typically won't pass test and stub directories together. This should have either two test directoreis or two stub directores.

StarKhan6368 replied:
Thanks, I've renamed these folders to be generic to avoid this confusion.

StarKhan6368 committed:
Ensure property cleanup, fix test/stub confusion.

- Properties to be cleared in finally block.
- Rename example folders to generics to avoid
  confusion.

joelrosario reviewed (approved):
LGTM
